### PR TITLE
Implement error attachments v2

### PR DESCRIPTION
**Pull request by guperrot**

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -53,8 +53,8 @@ var ANDROID_ASSEMBLIES_FOLDER = TEMPORARY_PREFIX + "AndroidAssemblies";
 var PCL_ASSEMBLIES_FOLDER = TEMPORARY_PREFIX + "PCLAssemblies";
 
 // Native SDK versions
-var ANDROID_SDK_VERSION = "0.7.0";
-var IOS_SDK_VERSION = "0.7.0";
+var ANDROID_SDK_VERSION = "0.8.1-2+06d8a1f";
```
Comment (dhei) [inline build.cake]: Can you explain why android sdk version contains a commit hash while ios doesn't?

Comment (guperrot) [inline build.cake]: On Android, we don't create a git tag/release when we just want an incremental build. We add the hash to know where this build is coming from. .NET has this too.

File: SDK/MobileCenterCrashes/Microsoft.Azure.Mobile.Crashes.iOS/PlatformCrashes.cs [hunk 2]
```diff
@@ -16,7 +15,7 @@ class PlatformCrashes : PlatformCrashesBase
         public override SentErrorReportEventHandler SentErrorReport { get; set; }
         public override FailedToSendErrorReportEventHandler FailedToSendErrorReport { get; set; }
         public override ShouldProcessErrorReportCallback ShouldProcessErrorReport { get; set; }
-        //public override GetErrorAttachmentCallback GetErrorAttachment { get; set; }
+        public override GetErrorAttachmentsCallback GetErrorAttachments { get; set; }
         public override ShouldAwaitUserConfirmationCallback ShouldAwaitUserConfirmation { get; set; }
 
         CrashesDelegate crashesDelegate { get; set; }
```
Comment (achocron) [inline SDK/MobileCenterCrashes/Microsoft.Azure.Mobile.Crashes.iOS/PlatformCrashes.cs]: This property has different casing from others, should be called `CrashesDelegate`, right?

File: SDK/MobileCenterCrashes/Microsoft.Azure.Mobile.Crashes.iOS/CrashesDelegate.cs [hunk 3]
```diff
@@ -23,22 +23,34 @@ public override bool CrashesShouldProcessErrorReport(MSCrashes crashes, MSErrorR
             return _owner.ShouldProcessErrorReport(report);
```
Comment (achocron) [inline SDK/MobileCenterCrashes/Microsoft.Azure.Mobile.Crashes.iOS/CrashesDelegate.cs]: Could this throw a null reference exception in an edge case?

File: SDK/MobileCenterCrashes/Microsoft.Azure.Mobile.Crashes.iOS/PlatformCrashes.cs (hunk 2 above)
Comment (guperrot) [inline SDK/MobileCenterCrashes/Microsoft.Azure.Mobile.Crashes.iOS/PlatformCrashes.cs]: No

File: SDK/MobileCenterCrashes/Microsoft.Azure.Mobile.Crashes.iOS/PlatformCrashes.cs [hunk 4]
```diff
@@ -16,7 +15,7 @@ class PlatformCrashes : PlatformCrashesBase
         public override SentErrorReportEventHandler SentErrorReport { get; set; }
```
Comment (guperrot) [inline SDK/MobileCenterCrashes/Microsoft.Azure.Mobile.Crashes.iOS/PlatformCrashes.cs]: My pr is not touching this, can we discuss that offline?

File: SDK/MobileCenterCrashes/Microsoft.Azure.Mobile.Crashes.iOS/PlatformCrashes.cs (hunk 2 above)
Comment (guperrot) [inline SDK/MobileCenterCrashes/Microsoft.Azure.Mobile.Crashes.iOS/PlatformCrashes.cs]: Ok but my pr is not touching this.

File: SDK/MobileCenterCrashes/Microsoft.Azure.Mobile.Crashes.iOS/CrashesDelegate.cs (hunk 3 above)
Comment (guperrot) [inline SDK/MobileCenterCrashes/Microsoft.Azure.Mobile.Crashes.iOS/CrashesDelegate.cs]: We can look at it but this is not related to this pr.
